src/Storage/PathBuilder/BasePathBuilder.php

```diff
@@ -214,6 +215,22 @@ protected function _buildFilename(EntityInterface $entity, array $options = [])
 	}
 
 /**
+ * Checks if a subdirectory has been set in the adapter settings.
+ *
+ * Used to add to the URL functions.
+ *
+ * @param string $adapter Adapter config name set in the StorageManager
+ * @return string
+ */
+	protected function _adapterSubDirectory($adapter) {
```

Comment:
This is specific to the S3 builder and should go into the S3 builder and *not* in the generic base.